File: WebApp/src/interfaces/unitInterfaces/Unit.ts
import type { PagedResponse } from '../PagedResponse';

export type EmergencyServiceType = 'POLICIA' | 'SAMU' | 'BOMBEIROS';
export type UnitStatus = 'DISPONIVEL' | 'RESERVADA' | 'DESLOCAMENTO' | 'EM_ATENDIMENTO' | 'INDISPONIVEL';

export interface EmergencyService {
  id: number;
  type: EmergencyServiceType;
  emergencyNumber: string;
  displayName: string;
}

export interface Unit {
  id: number;
  name: string;
  service: EmergencyService;
  postalCode: string | null;
  street: string | null;
  number: string | null;
  complement: string | null;
  neighborhood: string | null;
  city: string | null;
  state: string | null;
  reference: string | null;
  address: string;
  status: UnitStatus;
  createdAt: string;
  updatedAt: string;
}

export interface UnitPayload {
  name: string;
  service: EmergencyServiceType;
  postalCode: string;
  street: string;
  number: string;
  complement?: string;
  neighborhood: string;
  city: string;
  state: string;
  reference?: string;
  status: UnitStatus;
  expectedUpdatedAt?: string;
}

export interface UnitFilters {
  search: string;
  emergencyService: '' | EmergencyServiceType;
  status: '' | UnitStatus;
}

export type UnitsPagination = PagedResponse<Unit>;

File: Api/Models/Unit.cs
using Api.Models.Common;
using Api.Models.Enums;

namespace Api.Models;

public sealed class Unit : AuditableEntity
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public string NormalizedName { get; set; } = null!;
    public int EmergencyServiceId { get; set; }
    public EmergencyService EmergencyService { get; set; } = null!;
    public string? PostalCode { get; set; }
    public string? Street { get; set; }
    public string? Number { get; set; }
    public string? Complement { get; set; }
    public string? Neighborhood { get; set; }
    public string? City { get; set; }
    public string? State { get; set; }
    public string? AddressReference { get; set; }
    public string? GeocodingSource { get; set; }
    public decimal Latitude { get; set; }
    public decimal Longitude { get; set; }
    public UnitStatus Status { get; set; } = UnitStatus.DISPONIVEL;
    public ICollection<Dispatch> Dispatches { get; set; } = new List<Dispatch>();
    public User? OperationalUser { get; set; }
}
